CMakeLists.txt

```diff
@@ -1,134 +1,87 @@
-set(CPACK_SOURCE_GENERATOR "TGZ")
-set(CPACK_SOURCE_PACKAGE_FILE_NAME
-  "${CMAKE_PROJECT_NAME}-${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}")
-set(CPACK_SOURCE_IGNORE_FILES
-  "/build/;/.git/;/.github/;/*.patch;/.bundle/;/_site/;/vendor/;~$;${CPACK_SOURCE_IGNORE_FILES}")
-include(CPack)
-
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-
-message(STATUS "pgmoneta ${VERSION_STRING}")
-
-include(CheckCCompilerFlag)
-include(CheckCSourceCompiles)
-include(FindPackageHandleStandardArgs)
-include(GNUInstallDirs)
-
-if (NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: Debug Release" FORCE)
-endif ()
-
-message(STATUS "Build type is ${CMAKE_BUILD_TYPE}")
-message(STATUS "System is ${CMAKE_SYSTEM_NAME}")
-
-set(SUPPORTED_COMPILERS "GNU" "Clang" "AppleClang")
-
-# Check for a supported compiler
-if (NOT CMAKE_C_COMPILER_ID IN_LIST SUPPORTED_COMPILERS)
-   message(FATAL_ERROR "Unsupported compiler ${CMAKE_C_COMPILER_ID}. Supported compilers are: ${SUPPORTED_COMPILERS}")
-endif ()
-
-CHECK_C_COMPILER_FLAG("-std=c17" COMPILER_SUPPORTS_C17)
-if(NOT COMPILER_SUPPORTS_C17)
-  message(FATAL_ERROR "The compiler ${CMAKE_C_COMPILER} has no C17 support. Please use a different C compiler.")
-endif()
-
-find_package(ZLIB)
-if (ZLIB_FOUND)
-  message(STATUS "zlib found")
-else ()
-  message(FATAL_ERROR "zlib needed")
-endif()
-
-find_package(Zstd)
-if (ZSTD_FOUND)
-  message(STATUS "zstd found")
-else ()
-  message(FATAL_ERROR "zstd needed")
-endif()
-
-find_package(Lz4)
-if (LZ4_FOUND)
-  message(STATUS "lz4 found")
-else ()
-  message(FATAL_ERROR "lz4 needed")
-endif()
-
-find_package(Libev 4.11)
-if (LIBEV_FOUND)
-  message(STATUS "libev found")
-else ()
-  message(FATAL_ERROR "libev needed")
-endif()
-
-find_package(OpenSSL)
-if (OPENSSL_FOUND)
-  message(STATUS "OpenSSL found")
-else ()
-  message(FATAL_ERROR "OpenSSL needed")
-endif()
-
-find_package(Rst2man)
-if (RST2MAN_FOUND)
-  message(STATUS "rst2man found")
-else ()
-  message(FATAL_ERROR "rst2man needed")
-endif()
-
-find_package(Libssh)
-if (LIBSSH_FOUND)
-  message(STATUS "libssh found")
-else ()
-  message(FATAL_ERROR "libssh needed")
-endif()
-
-find_package(Libcurl)
-if (LIBCURL_FOUND)
-  message(STATUS "libcurl found")
-else ()
-  message(FATAL_ERROR "libcurl needed")
-endif()
-
-if (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
-  find_package(Libatomic)
-  if (LIBATOMIC_FOUND)
-    message(STATUS "libatomic found")
-  else ()
-    message(FATAL_ERROR "libatomic needed")
-  endif()
-
-  find_package(Systemd)
-  if (SYSTEMD_FOUND)
-    message(STATUS "systemd found")
-  else ()
-    message(FATAL_ERROR "systemd needed")
-  endif()
-endif()
-
-file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/src/")
-
-add_subdirectory(doc)
-add_subdirectory(src)
+cmake_minimum_required(VERSION 3.24)
```

Comment:
This is an unrelated change

Reply:
Remove the `CMakeLists.txt` change, then squash and force push

Reply:
Done. So sorry about that, I'm using CLion locally and it won't let me use ctrl + click to jump to definitions unless it configures the cmakelist, which is based on linux environment. So I had to make my own cmakelist and forgot to restore it when pushing.

Reply:
https://www.scraggo.com/how-to-squash-commits/

Reply:
Done, I thought I just need to squash the last 2 commits :), the cmakelist fix and the elaboration one.